### ISSUE TITLE
Heavy layout reflow on drag start and end

### ISSUE THREAD

**Issue by darkowic** [atlassian/react-beautiful-dnd#1850]
### Expected behavior

Minimize application reflow when lifting, dragging and dropping an item.

### Actual behavior

Tests performed on https://react-beautiful-dnd.netlify.app/?path=/story/board--long-lists-in-a-short-container (in dev mode for sourcemaps support). **I applied 6 times CPU throttling** (i5 6500K - 4.5GHz)

Before lifting an item, there is **a huge layout reflow triggered**:
![reflow](https://user-images.githubusercontent.com/11510581/83072644-6be1de00-a06f-11ea-9f71-2b8f8003e4f7.png)

My curiosity lead me to this mysterious hidden element in the style marshal
https://github.com/atlassian/react-beautiful-dnd/blob/master/src/view/use-style-marshal/use-style-marshal.js#L36
![reflow2](https://user-images.githubusercontent.com/11510581/83072955-ea3e8000-a06f-11ea-991c-c37be332568e.png)
**Looks like its `textContent` is modified and it causes application reflow.** After commenting it out, the layout reflow is gone:
![no-reflow](https://user-images.githubusercontent.com/11510581/83072970-f0346100-a06f-11ea-95da-5e38c4950208.png)

If is significant in application which I am building (quite complex board):
![reflow-my-app](https://user-images.githubusercontent.com/11510581/83073233-6042e700-a070-11ea-9eb2-6ecf6361866b.png)
![reflow-my-app-2](https://user-images.githubusercontent.com/11510581/83073130-338ecf80-a070-11ea-8835-2bf860284242.png)
Again, commenting it out improves is drastically:
![no-reflow-my-app](https://user-images.githubusercontent.com/11510581/83073118-2eca1b80-a070-11ea-9486-997d8966a12f.png)

### Steps to reproduce

^ See above. Try to throttle your CPU and try to move element in some big boards.

### Suggested solution?

If we could avoid adding this element to dom and modifying it we could get noticeable performance boost especially on slower devices. May I know why it is required?

### What version of `React` are you using?
^16.13.1

### What version of `react-beautiful-dnd` are you running?

Latest - v13

### What browser are you using?

Chrome 83

### Demo

Not needed - see the actuall behaviour.

**Comment by schichulin** [comment]
I faced exactly the same issue in my application where I have a table with 600 **not draggable** rows with pretty heavy content inside AND a popup with a few (up to 20) draggables. The popup is rendered in the `document.body` and has nothing in common with the table.

BEFORE
![image](https://user-images.githubusercontent.com/27777649/131827544-4c57422e-8ccf-40f1-9733-561bf8c9d71c.png)

https://user-images.githubusercontent.com/27777649/131828948-aeecf62c-2107-4c35-ba44-55729d1ae9bd.mov


AFTER
![image](https://user-images.githubusercontent.com/27777649/131828707-c8d71bb3-374d-4aa1-af06-d135affbcae3.png)

https://user-images.githubusercontent.com/27777649/131829081-92c0be26-8202-4f36-b061-b290db67a5cb.mov

by commenting exactly this line the lags are gone, but the animations stopped working as well.

**Comment by kennyzli** [comment]
I have exact the issue, looking forward to see the solution!!!!

**Comment by joseffsmith** [comment]
Yep just want to add that this is something I've faced as well. I traced it back to the `textContent` line as well.

Dropping in a debugger here let me see that the textContent being set is essentially a stylesheet that is being appended to the `<head>` of the document.

![image](https://user-images.githubusercontent.com/26124821/136002886-65575958-914e-49d4-ac8f-706b01796ccf.png)

So whenever a drag is started the stylesheet is modified and because styles are applied to the `body` I assume it triggers a style recalculation for every element inside the `body`.

This is especially noticeable on our side because we have some pretty hefty svg's with a few thousand elements.

I feel like these styles could be applied somewhere close to the `DragDropContext` instead of the `body` which would help isolate the style recalculation hopefully

**Comment by joseffsmith** [comment]
I think I've worked out that it's the `user-select` that's causing the style reflow.

I've modified the code locally and by moving the `user-select` style onto the `draggable` element it prevents the style reflow for all elements in the document.

You can see here in the MDN docs that clicking on an element with `user-select: none` and trying to highlight any other elements doesn't work, so applying the `user-select: none` to only the draggable element will still have the intended effect.
https://developer.mozilla.org/en-US/docs/Web/CSS/user-select#result

Here is the code for the PR that I'm going to open, it'll be my first contribution to an open source project so apologies if I missed something.

![image](https://user-images.githubusercontent.com/26124821/136020518-e8d32acc-8dd2-422d-a267-fccaf2ea5596.png)
